Remember data format after image or sound import

cwillisf opened the pull request:
### Resolves

Resolves LLK/scratch-gui#407

### Proposed Changes

When importing an asset, record its actual data format (file type / extension) instead of the `AssetType` (runtime data type).

### Reason for Changes

We need to know if an asset is (for example) a PNG or JPG when attempting to load it, but the `AssetType` only expresses the default.

### Test Coverage

This PR includes a new test which verifies that asset details survive a round trip through SB3 serialization & deserialization. In support of this new test, and to correct an existing misuse, the `attachTestStorage` function has been replaced with `makeTestStorage`, which now simply returns a prepared storage instance instead of automatically attaching it to a VM instance.

Commit:
Remember the data format when an asset is imported

Commit:
Convert `attachTestStorage` to `makeTestStorage`

Rather than assuming that the storage instance will be attached to a VM,
just return it. Callers may attach it to a `VM` or (in the case of
`import_sb2.js`) to a `Runtime`.

Commit:
Add test for SB3 serialization round trip

cwillisf commented on test/integration/import_sb2.js:
```diff
@@ -20,7 +20,7 @@ test('default', t => {
 
     // Create runtime instance & load SB2 into it
     const rt = new runtime();
-    attachTestStorage(rt);
```
Strictly speaking, this call is incorrect: `attachTestStorage` expects a VM, not a runtime. It was working anyway because both `VirtualMachine` and `Runtime` have a method called `attachStorage` and that's all `attachTestStorage` was calling, but IMO moving the attachment out is safer and simpler than making `attachTestStorage` officially accept both types.

cwillisf commented on test/integration/sb3-roundtrip.js:
```diff
@@ -0,0 +1,110 @@
+        // `Clone.simple` here helps in two ways:
+        // 1. it ensures that any non-serializable data is thrown away, and
+        // 2. `sb3.deserialize` does some `hasOwnProperty` checks which fail on the object returned by `sb3.serialize`
+        //    due to its inheritance structure.
```
It took me far longer than it should have to figure out that this was a problem...

rschamp replied:
What inheritance structure specifically caused this issue?

Commit:
Clarify "flattening" step in SB3 roundtrip test

Commit:
Add support for older SB3 JSON

If the `dataFormat` field is absent, fall back on the older `assetType`
field. That can at least help us tell if something is a vector or bitmap
image, though it doesn't tell us which kind of bitmap.

rschamp reviewed (approved):
The backwards compatibility works! Looks great, thanks!
